.github/workflows/micronaut-gradle.yml

```diff
@@ -481,11 +483,33 @@ jobs:
 
       - name: Output pushed image tag
         id: output-pushed-image
-        if: inputs.publish-image == 'true'
+        if: inputs.publish-image && !inputs.lambda-build
         shell: bash
         run: |
           echo "::set-output name=image-tag::${{ needs.setup.outputs.image-name }}:${{ steps.build-and-push.outputs.image-version }}"
 
+      - name: Extract lambda from image
+        id: extract-lambda
+        if: inputs.lambda-build
+        shell: bash
+        run: |
+          filename=app-${{ needs.setup.outputs.version }}-lambda.zip
+          container_id=fn
+
+          docker create --name "$container_id" "${{ steps.build-and-push.outputs.image-id }}"
+          docker cp "$container_id:/function/function.zip" "$filename"
+          docker rm -v "$container_id"
+
+          echo "::set-output name=filename::$filename"
+
+      - name: Upload lambda artifact
+        uses: actions/upload-artifact@v3
+        if: inputs.lambda-build
+        with:
+          name: ${{ steps.extract-lambda.outputs.filename }}
+          path: ${{ steps.extract-lambda.outputs.filename }}
+          retention-days: 1
```

Comment:
J'me demande, il y aurait moyen quand on fait une release de prendre le zip et de l'ajouter à la release?

Reply:
Je l'ai mit dans le workflow de la lambda direct. Ça peut sûrement être ramené ici

https://github.com/equisoft/github-webhook-to-action-bridge/blob/main/.github/workflows/app.yml#L30